jlibra-core/src/main/java/dev/jlibra/mnemonic/ExtendedPrivKey.java

```diff
@@ -36,18 +38,35 @@ public String getAddress() {
         SHA3.DigestSHA3 digestSHA3 = new SHA3.Digest256();
         byte[] saltDigest = digestSHA3.digest("RawTransaction@@$$LIBRA$$@@".getBytes());
         byte[] transactionBytes = rawTransaction.toByteArray();
-        byte[] message = ByteBuffer.allocate(saltDigest.length + transactionBytes.length)
+        byte[] messageBytes = ByteBuffer.allocate(saltDigest.length + transactionBytes.length)
                 .put(saltDigest)
                 .put(transactionBytes)
                 .array();
+        byte[] message = digestSHA3.digest(messageBytes);
 
+        return sign(message);
+    }
+
+    @VisibleForTesting
+    byte[] sign(byte[] message) {
         try {
             Ed25519Signer signer = new Ed25519Signer();
             signer.init(true, new Ed25519PrivateKeyParameters(privateKey.getData(), 0));
-            signer.update(digestSHA3.digest(message), 0, message.length);
+            signer.update(message, 0, message.length);
```

Comment:
Nice, that was an error I ran into, message was hashed, but message length was the original (unhashed) array. I actually set the message length to 32 bytes, as that's the only possibility, but message.length is fine, as hash algo might change.